lib/filetests/src/runner.rs

```diff
@@ -329,12 +341,24 @@ impl TestRunner {
     }
 
     /// Scan pushed directories for tests and run them.
-    pub fn run(&mut self) -> TestResult {
+    pub fn run(&mut self, passes: Option<&Vec<String>>, target: Option<&str>) -> TestResult {
         let started = time::Instant::now();
-        self.scan_dirs();
-        self.schedule_jobs();
-        self.drain_threads();
-        self.report_slow_tests();
+
+        // If passes are present from command line, then no threads.
+        match passes {
+            Some(pass_vec) => {
```

Comment:
Since the two cases are different enough, and there's not much before/after, plus it makes all the other `run()` call sites look a bit weird (because of the `None` argument, which doesn't provide much information about what type it is, what should be provided, etc.), maybe it could be split into a different function `run_passes(&mut self, passes: &Vec<String>, target: &str)`?